types/beyonk__gdpr-cookie-consent-banner/index.d.ts

```diff
@@ -0,0 +1,33 @@
+// Type definitions for @beyonk/gdpr-cookie-consent-banner 9.0
+// Project: https://github.com/beyonk-adventures/gdpr-cookie-consent-banner
+// Definitions by: Ben Grynhaus <https://github.com/bengry>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface Choice {
+    label: string;
+    description: string;
+    value: boolean;
+}
+
+export type Category = 'necessary' | 'tracking' | 'analytics' | 'marketing';
+
+export interface Options {
+    cookieName?: string;
+    cookieConfig?: {
+        domain: string;
+        path: string;
+    };
+    heading?: string;
+    description?: string;
+    acceptLabel?: string;
+    rejectLabel?: string;
+    settingsLabel?: string;
+    closeLabel?: string;
+    choices?: Partial<Record<Category, Choice | false>>;
+    showEditIcon?: boolean;
+    categories?: Partial<Record<Category, () => void>>;
+}
+
+declare function attachBanner(bodyElement: HTMLElement, options?: Options): void;
+
+export default attachBanner;
```

Comment:
you should consder using 'export as namespace' to support global script usage:
https://github.com/beyonk-adventures/gdpr-cookie-consent-banner#using-raw-javascript
https://www.typescriptlang.org/docs/handbook/modules.html#math-libdts

Reply:
Done in latest revision. I'm not sure if I can do that in one declaration file. If its possible please LMK and I'll change it.